### PR TITLE
BUGFIX: Focus first form element in node creation dialog

### PR DESCRIPTION
**Pull request by markusguenther**
**What I did**
The NodeCreationDialog always throws errors because the options are passed thru as array instead of an object. So I modified the data type here.

The already set autoFocus property did not work for the first element of the Creation Dialog.
So this patch also fixes the setting of the autofocus for the editor fields.

Another issue was that the dialog always gets the focus by default and therefore the first element was not focused.

**How to verify it**

Create a new page in the backend and the title field will be focused.

Fixes: #2696

### CODE REVIEW

File: packages/react-ui-components/src/Dialog/dialog.tsx [hunk 1]
```diff
@@ -58,6 +58,11 @@ export interface DialogProps {
      */
     readonly actions: ReadonlyArray<ReactNode>;
 
+    /**
+     * This prop controls the focus state of the Dialog.
+     */
+    readonly autoFocus: boolean;
```
Comment (dimaip) [inline packages/react-ui-components/src/Dialog/dialog.tsx:64]: How does this differ from `setFocus` prop that we already have?

Comment (dimaip) [reply packages/react-ui-components/src/Dialog/dialog.tsx:64]: nevermind, was looking at the wrong component....

File: packages/react-ui-components/src/TextInput/textInput.tsx [hunk 2]
```diff
@@ -52,11 +52,16 @@ export interface TextInputProps extends Omit<React.InputHTMLAttributes<HTMLInput
...
+
+    constructor(props: any) {
+        super(props);
+        this.ref = React.createRef();
+    }
 
     public readonly componentDidMount = (): void => {
-        if (this.ref && this.props.setFocus) {
-            this.ref.focus();
+        if (this.ref && this.ref.current && (this.props.setFocus || this.props.autoFocus)) {
```
Comment (dimaip) [inline packages/react-ui-components/src/TextInput/textInput.tsx:63]: Where is `autoFocus` prop coming from? It's not defined in proptypes. How does it differ from `setFocus`?

Comment (markusguenther) [reply packages/react-ui-components/src/TextInput/textInput.tsx:63]: The NodeCreationDialog sets the property since some years ;)

Comment (markusguenther) [reply packages/react-ui-components/src/TextInput/textInput.tsx:63]: TextInput also has the prop for some time.

Comment (dimaip) [reply packages/react-ui-components/src/TextInput/textInput.tsx:63]: Ah, you mean to say that it's is just passed down to the DOM element as a DOM attribute here? https://github.com/neos/neos-ui/blob/72caa0aa8a97ed79ab37a9c701136e8b6405f125/packages/react-ui-components/src/TextInput/textInput.tsx#L107
Didn't think about it at first, good point then.

**Pull request merged**